sopel/coretasks.py

```diff
@@ -835,7 +835,7 @@ def auth_proceed(bot, trigger):
     else:
         return
     sasl_username = sasl_username or bot.nick
-    sasl_token = r'\0'.join((sasl_username, sasl_username, sasl_password))
+    sasl_token = '\0'.join((sasl_username, sasl_username, sasl_password))
```

Comment:
Why not do it in a non-deprecated way?

```suggestion
    sasl_token = '\x00'.join((sasl_username, sasl_username, sasl_password))
```

Reply:
TIL.